### ISSUE TITLE
On Code Formatting

### ISSUE THREAD

**Issue by Fallenhh** [chipsalliance/dromajo#37]
As mentioned in the #35  , I think there are some ways to format the codes from different contributors.

- There are some out-of-the-box [plugins](https://github.com/marketplace/actions/clang-format-lint) to format the c code on the actions marketplace.
- It would also be a good idea to include the standard guidelines for formatting in the README.md and add the .clang-format file to the repo.

**Comment by et-tommythorn** [comment]
Yes. The convention Jose and I agreed on is to require `clang-format {src,include/*` but better documentation and automation would go a long way.  Looking for volunteers.
